mlabs/src/Mlabs/NFT/Contract.hs

```diff
@@ -1,373 +1,221 @@
-              (paidToOwner, paidToAuthor) = calculateShares bid $ dNft'share oldDatum
-              newValue = ciTxOut ^. ciTxOutValue
-              (lookups, tx) =
-                ( mconcat
-                    [ Constraints.unspentOutputs userUtxos
-                    , Constraints.typedValidatorLookups txPolicy
-                    , Constraints.mintingPolicy nftPolicy'
-                    , Constraints.otherScript (validatorScript txPolicy)
-                    , Constraints.unspentOutputs $ Map.singleton nftOref ciTxOut
-                    ]
-                , mconcat
-                    [ Constraints.mustPayToTheScript newDatum' newValue
-                    , Constraints.mustIncludeDatum newDatum
-                    , Constraints.mustPayToPubKey (getUserId . dNft'owner $ oldDatum) paidToOwner
-                    , Constraints.mustPayToPubKey (getUserId . dNft'author $ oldDatum) paidToAuthor
-                    , Constraints.mustSpendScriptOutput
-                        nftOref
-                        (Redeemer . PlutusTx.toBuiltinData $ action)
-                    ]
-                )
-          void $ Contract.submitTxConstraintsWith @NftTrade lookups tx
-          void $ Contract.logInfo @Hask.String $ printf "Bought %s" $ Hask.show val
-
--- SET PRICE --
-setPrice :: SetPriceParams -> Contract w NFTAppSchema Text ()
-setPrice spParams = do
-  result <-
-    Contract.runError $ do
-      (oref, ciTxOut, datum) <- findNft txScrAddress $ sp'nftId spParams
-      runOffChainChecks datum
-      let (tx, lookups) = mkTxLookups oref ciTxOut datum
-      ledgerTx <- Contract.submitTxConstraintsWith @NftTrade lookups tx
-      void $ Contract.awaitTxConfirmed $ Ledger.txId ledgerTx
-  either Contract.logError (const $ Contract.logInfo @Hask.String "New price set") result
-  where
-    mkTxLookups oref ciTxOut datum =
-      let newDatum = datum {dNft'price = sp'price spParams}
-          redeemer = asRedeemer $ SetPriceAct (sp'price spParams) $ nftCurrency (dNft'id datum)
-          newValue = ciTxOut ^. ciTxOutValue
-          lookups =
-            mconcat
-              [ Constraints.unspentOutputs $ Map.singleton oref ciTxOut
-              , Constraints.typedValidatorLookups txPolicy
-              , Constraints.otherScript (validatorScript txPolicy)
-              ]
-          tx =
-            mconcat
-              [ Constraints.mustSpendScriptOutput oref redeemer
-              , Constraints.mustPayToTheScript newDatum newValue
-              ]
-       in (tx, lookups)
 
-    runOffChainChecks :: DatumNft -> Contract w NFTAppSchema Text ()
-    runOffChainChecks datum = do
-      ownPkh <- pubKeyHash <$> Contract.ownPubKey
-      if isOwner datum ownPkh
-        then pure ()
-        else Contract.throwError "Only owner can set price"
-      if priceNotNegative (sp'price spParams)
-        then pure ()
-        else Contract.throwError "New price can not be negative"
+-- buy :: BuyRequestUser -> UserContract ()
```

Comment:
I don't think we need to leave so many comments with legacy code all over the place.